### PR TITLE
Add --no-extra flag to join command

### PR DESCRIPTION
**Pull request by rgee0**
## Why do you need this?
It was found that when joining servers to a cluster any additional servers would not honour the `--no-extras` flag used with the install command.

It is necessary, therefore, for the flag to be added to `join` and only be applied in situations where `--server` has been provided.

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

Although closed, #408 refers

## Description

Adds `--no-extras` to the join command such that it will work when used with `--server`.

### Add new flag without --server
```
$ ./k3sup-darwin join \             
  --ip $SERVER_IP_2 \
  --user $USER \
  --server-user $USER \
  --server-ip $SERVER_IP \
  --no-extras --print-command

Error: --no-extras can only be used with --server
```
### Add new flag with --server
```
 $ ./k3sup-darwin join \
  --ip $SERVER_IP_2 \
  --user $USER \
  --server-user $USER \
  --server-ip $SERVER_IP \
  --no-extras --print-command --server

Running: k3sup join
Joining 209.97.141.169 => 138.68.158.53
ssh: sudo cat /var/lib/rancher/k3s/server/node-token

Received node-token from 138.68.158.53.. ok.
ssh: curl -sfL https://get.k3s.io | K3S_URL='https://138.68.158.53:6443'  \
K3S_TOKEN='<token>::server:<alphanumeric>' \
INSTALL_K3S_CHANNEL='stable' \
INSTALL_K3S_EXEC='server --server https://138.68.158.53:6443' \
sh -s -  --disable servicelb --disable traefik
```
Note the addition of ` --disable servicelb --disable traefik` to the output command

### With `--tls-san` added as this was in the area of change:
```
$ ./k3sup-darwin join \
  --ip $SERVER_IP_2 \
  --user $USER \
  --server-user $USER \
  --server-ip $SERVER_IP \
  --no-extras --print-command --server  --tls-san 127.0.0.1
Running: k3sup join
Joining 209.97.141.169 => 138.68.158.53
ssh: sudo cat /var/lib/rancher/k3s/server/node-token

Received node-token from 138.68.158.53.. ok.
ssh: curl -sfL https://get.k3s.io | K3S_URL='https://138.68.158.53:6443' K3S_TOKEN='<alphanumeric>::server:<alphanumeric>' INSTALL_K3S_CHANNEL='stable' INSTALL_K3S_EXEC='server --server https://138.68.158.53:6443 --tls-san 127.0.0.1' sh -s -  --disable servicelb --disable traefik
```
### With `--tls-san` and without `--server`:
```
./k3sup-darwin join \
  --ip $SERVER_IP_2 \
  --user $USER \
  --server-user $USER \
  --server-ip $SERVER_IP \
  --print-command  --tls-san 127.0.0.1 
Error: --tls-san can only be used with --server
```
Demonstrates previous behaviour unchanged

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by alexellis — APPROVED** [review]
LGTM. Thank you for helping to fix a confusing bug 👍

**Comment by alexellis** [comment]
Could you check whether `--no-extras` gets correctly merged with anything passed via `--k3s-extra-args`?

**Comment by rgee0** [comment]
> Could you check whether `--no-extras` gets correctly merged with anything passed via `--k3s-extra-args`?

Updated the test output above with `Additional Test case using --no-extras with --k3s-extra-args set:`

TL;DR is that it merges the values from the `--no-extras` flag correctly

**Comment by alexellis** [comment]
Thanks @rgee0
